migrations/versions/0a6b82b55983_.py

```diff
@@ -0,0 +1,84 @@
+"""empty message
+
+Revision ID: 0a6b82b55983
+Revises: 3ee58dee57c9
+Create Date: 2018-09-04 19:09:45.866336
+
+"""
+from alembic import op
+import sqlalchemy as sa
+import re
+from server.models.postgis.message import Message, MessageType
+
+
+# revision identifiers, used by Alembic.
+revision = '0a6b82b55983'
+down_revision = '3ee58dee57c9'
+branch_labels = None
+depends_on = None
+
+messages = Message.__table__
+
+def upgrade():
+    conn = op.get_bind()
+
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column('messages', sa.Column('message_type', sa.Integer(), nullable=True))
+    op.add_column('messages', sa.Column('project_id', sa.Integer(), nullable=True))
+    op.add_column('messages', sa.Column('task_id', sa.Integer(), nullable=True))
+    op.create_foreign_key('fk_message_projects', 'messages', 'projects', ['project_id'], ['id'])
+    op.create_index(op.f('ix_messages_message_type'), 'messages', ['message_type'], unique=False)
+    op.create_index(op.f('ix_messages_project_id'), 'messages', ['project_id'], unique=False)
+    op.create_index(op.f('ix_messages_task_id'), 'messages', ['task_id'], unique=False)
+    # ### end Alembic commands ###
+
+    project_task_re = re.compile('project/(\d+)/\?task=(\d+)')
+    validated_notification_re = re.compile('been validated')
+    invalidated_notification_re = re.compile('been marked invalid')
+    mention_notification_re = re.compile('You were mentioned')
+    welcome_re = re.compile('Welcome to the HOT Tasking Manager')
+    message_all_re = re.compile('project/(\d+)\?tab=chat')
+
+    # Attempt to classify existing messages
+    for message in conn.execute(messages.select()):
+        message_type = None
+        project_id = None
+        task_id = None
+
+        match = project_task_re.search(message.subject)
+        if match:
+            project_id = match.group(1)
+            task_id = match.group(2)
+
+        if validated_notification_re.search(message.subject):
+            message_type = MessageType.VALIDATION_NOTIFICATION.value
+        elif invalidated_notification_re.search(message.subject):
+            message_type = MessageType.INVALIDATION_NOTIFICATION.value
+        elif mention_notification_re.search(message.subject):
+            message_type = MessageType.MENTION_NOTIFICATION.value
+        elif welcome_re.search(message.subject):
+            message_type = MessageType.SYSTEM.value
+
+        # Look for direct messages from project managers
+        if message_type is None and project_id is None:
+            match = message_all_re.search(message.message)
+            if match:
+                project_id = match.group(1)
+                message_type = MessageType.DIRECT.value
```

Comment:
It looks like this `DIRECT` type hasn't been defined: https://github.com/hotosm/tasking-manager/pull/1423/files#diff-c7e71dbcd063d5f165d06ef62273f1b0R10